x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/event/AnalyticsEvent.java

```diff
@@ -41,44 +45,9 @@ public class AnalyticsEvent implements Writeable, ToXContentObject {
     public static final ParseField DATA_STREAM_TYPE_FIELD = new ParseField("type");
     public static final ParseField DATA_STREAM_NAMESPACE_FIELD = new ParseField("namespace");
     public static final ParseField DATA_STREAM_DATASET_FIELD = new ParseField("dataset");
-
-    /**
-     * Analytics context. Used to carry information to parsers.
-     */
-    public interface Context {
-        long eventTime();
-
-        Type eventType();
-
-        String eventCollectionName();
-
-        default AnalyticsCollection analyticsCollection() {
-            return new AnalyticsCollection(eventCollectionName());
-        }
-    }
-
-    static class AnalyticsEventContext implements Context {
-        private final AnalyticsEvent event;
-
-        AnalyticsEventContext(AnalyticsEvent event) {
-            this.event = Objects.requireNonNull(event, "event cannot be null");
-        }
-
-        @Override
-        public long eventTime() {
-            return event.eventTime();
-        }
-
-        @Override
-        public Type eventType() {
-            return event.eventType();
-        }
-
-        @Override
-        public String eventCollectionName() {
-            return event.eventCollectionName();
-        }
-    }
+    public static final ParseField CLIENT_ADDRESS_FIELD = new ParseField("client_address");
+    public static final ParseField USER_AGENT_FIELD = new ParseField("user_agent");
```

Comment:
Those fields should be declared in the SessionFieldParser since user_agent and ip are supposed to be part of the session.

BTW, the ip address field is supposed to be named session.ip instead of client_address.